test/rules/no-unnecessary-parens/test.ts.lint

```diff
@@ -0,0 +1,33 @@
+let x = (3);
```

Comment:
There's one edge case that would result in invalid code:
```ts
let y, z;
let x = (y = 1, z = 2);
```

I hope nobody writes such code.
Nontheless there should probably be an exception for `ts.SyntaxKind.BinaryExpression` where `operatorToken.kind === ts.SyntaxKind.CommaToken`

Reply:
Ugh, the comma operator is the worst. On my shortlist of new tslint rules is one to ban it entirely.

I made it so that it doesn't suggest a fix if the child uses a comma operator, but it still flags it as an issue. How does that sound?